File: Symulator Świata/sosnowsky.py
import nightshade as ns
import cybersheep
from animal import Animal
import random
GROW_CHANCE = 30


class Sosnowsky(ns.Nightshade):

    def __init__(self, world, coord=None):
        self.name = 's'
        self._strength = 10
        self._initiative = 0
        self._age = 0
        self.sprite = (125, 150, 25)
        self.world = world
        if coord is not None:
            self.world.board[coord] = self
        else:
            self.spawn()

    def action(self, coord):
        self.destroySides(coord)
        growRoll = random.randint(0, 100)
        if growRoll <= GROW_CHANCE:
            self.multiply(coord)

    def destroySides(self, coord):
        newCoord = coord
        newCoord.x += 1
        self.destroySide(newCoord)
        newCoord.x -= 2
        self.destroySide(newCoord)
        newCoord.x += 1
        newCoord.y -= 1
        self.destroySide(newCoord)
        newCoord.y += 2
        self.destroySide(newCoord)
        newCoord.y -= 1

    def destroySide(self, newCoord):
        if newCoord in self.world.board:
            if isinstance(self.world.board[newCoord], Animal):
                if not isinstance(self.world.board[newCoord], cybersheep.Cybersheep):
                    self.world.commentator.addComment("Barszcz sosnowskiego zabija {0}"
                                                      .format(self.world.board[newCoord].name))
                    self.world.board.pop(newCoord)

    def createChild(self, coord):
        s = Sosnowsky(self.world, coord)

File: PO PROJEKT 1/PO PROJEKT 1/PROJEKT PO PLIKI ŹRÓDŁOWE/Sosnowsky.cpp
#include "Sosnowsky.h"

Sosnowsky::Sosnowsky(World* world) {
	increaseStrength(10);
	setInitiative(0);
	sprite = 's';
	this->world = world;
	spawn();
}

Sosnowsky::Sosnowsky(World* world, Coordinate coord) {
	increaseStrength(10);
	setInitiative(0);
	sprite = 's';
	this->world = world;
	world->board.insert({ coord, this });
}

void Sosnowsky::action(Coordinate coord) {
	destroySides(coord);
	int growRoll = rand() % 100;
	if (growRoll <= GROW_CHANCE) {
		multiply(coord);
	}
}

void Sosnowsky::destroySides(Coordinate coord) {
	Coordinate newCoord = coord;
	newCoord.x += 1;
	if (world->board.count(newCoord) && dynamic_cast<Animal*>(world->board[newCoord])) {
		world->board.erase(newCoord);
	}
	newCoord.x -= 2;
	if (world->board.count(newCoord) && dynamic_cast<Animal*>(world->board[newCoord])) {
		world->board.erase(newCoord);
	}
	newCoord.x += 1;
	newCoord.y -= 1;
	if (world->board.count(newCoord) && dynamic_cast<Animal*>(world->board[newCoord])) {
		world->board.erase(newCoord);
	}
	newCoord.y += 2;
	if (world->board.count(newCoord) && dynamic_cast<Animal*>(world->board[newCoord])) {
		world->board.erase(newCoord);
	}
	newCoord.y -= 1;
}

Organism* Sosnowsky::createChild(Coordinate coord) {
	return new Sosnowsky(world, coord);
}

void Sosnowsky::draw() {
	cout << GREEN_BCKG << getSprite() << RESET;
}